app/services/replication_transfer_updater.rb

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) 2015 The Regents of the University of Michigan.
+# All Rights Reserved.
+# Licensed according to the terms of the Revised BSD License
+# See LICENSE.md for details.
+
+class ReplicationTransferUpdater
+
+  def self.update(record, params)
+    run_update(build_update(record,params))
+  end
+
+  def self.build_update(record, params)
+    [CancelUpdate, FixityValueUpdate, StoreRequestedUpdate, StoredUpdate].each do |update_class|
+      return update_class.new(record, params) if update_class.matching_update?(record, params)
+    end
+    return ReplicationTransferUpdate.new(record, params)
+  end
+
+  def self.run_update(update)
+    update.update
+  end
+
+  class ReplicationTransferUpdate
+    def initialize(record, params)
+      @record = record
+      @params = params
+    end
+    def update
+      @record.update(@params)
+    end
+    def we_requested?
+      @record.from_node&.namespace == Rails.configuration.local_namespace
+    end
+    def we_replicating?
+      @record.to_node&.namespace == Rails.configuration.local_namespace
+    end
+  end
+
+
+  class CancelUpdate < ReplicationTransferUpdate
+    def self.matching_update?(record, params)
+      record.respond_to?(:cancelled) &&
+        record.cancelled == false &&
+        params[:cancelled] == true
+    end
+    def update
+      @record.cancel!(@params[:cancel_reason])
+    end
+  end
+
+
+  class FixityValueUpdate < ReplicationTransferUpdate
+    def self.matching_update?(record, params)
+      record.respond_to?(:fixity_value) &&
+        record.fixity_value.nil? &&
+        params[:fixity_value] != nil
+    end
+
+    def update
+      result = @record.update(@params)
+      if result && we_requested?
+        request_storage
+      end
+      return result
+    end
+
+    private
+
+    def request_storage
+      if @params[:fixity_value] == get_digest.value
+        @record.update(store_requested: true)
+      else
+        @record.cancel!("fixity_reject")
+      end
+    end
+
+    def get_digest
+      @record.bag.message_digests.where(fixity_alg_id: @record.fixity_alg_id).first
+    end
```

Comment:
It seems that the `get_digest` method could include the `.value` call to simply return the fixity hash.  It would help if all these methods contained YARD documentation about `@return` types.  In this case, it seems to return a record and then `request_storage` has to use the `.value` accessor to get at the fixity value.  Moreover, that's all it ever does with the fixity record, so really what this class requires is a method to get that value, not the entire record.